.github/workflows/test.yml

```diff
@@ -149,10 +150,16 @@ jobs:
       - name: Checkout registry
         uses: ./.github/actions/checkout-registry
 
-      - name: Infra Tests
+      - name: Balance Tests
+        run: |
+          if git diff $PR_BASE HEAD --name-only | grep -qE '^typescript/infra/config/environments/mainnet3/balances|^.registryrc$'; then
+            yarn --cwd typescript/infra test:balance
+          fi
+
+      - name: Warp Config Tests
         run: |
-          if git diff --name-only | grep -qE '^typescript/infra/|^.registryrc$'; then
-            yarn --cwd typescript/infra test:infra
+          if git diff $PR_BASE HEAD --name-only | grep -qE '^typescript/infra/|^.registryrc$'; then
+            yarn --cwd typescript/infra test:balance
           fi
```

Comment:
if the PR branch has no changes to these files but you're out of date with HEAD which does have the changes, i think this still triggers a diff